### ISSUE TITLE
World Guard from FAWE doesn't work with Quests 3.5.7

### ISSUE THREAD

**Issue by SimianMussel92** [FlyingPikachu/Quests#599]
(__Required__) What is your Quests version / build number (do _not_ say "latest")?
The Quests version being used is 3.5.7. on a 1.13.2 server. The World Guard plugin is from FAWE (worldguard-legacy-7.0.0-SNAPSHOT-dist.jar). Thats the name of the World Guard plugin.

(__Required__) Please describe the problem and how to replicate it. Any console errors?
Quests seems to work just fine while not doing anything to do with World Guard. But, the console still seems to have some problems, but no erros. For example, Task 138 generated an exception. (I will paste a copy from the console bellow). More than anything, I think one can recreate the problem by just having FAWE World Edit, World Guard and Quests installed in a server. Although im not sure if having Quests from an earlier version would affect, im using Quests from a version for 1.12.

https://pastebin.com/C78wX41j (Here's the console when the server is starting up, I hope its helpful)

**Comment by FlyingPikachu** [comment]
The critical line from your startup log is:
>30.12 08:26:39 [Server] INFO java.lang.NoSuchMethodError: com.sk89q.worldguard.bukkit.WorldGuardPlugin.getRegionManager(Lorg/bukkit/World;)Lcom/sk89q/worldguard/protection/managers/RegionManager;

As far as I can tell, this is not an issue with FAWE specifically, but with recent beta builds of the official WorldGuard (WG) from which FAWE is based.

Currently, WG builds for Bukkit/Spigot are considered "legacy" as they move towards a more platform-independent structure. From [recent commits](https://github.com/sk89q/WorldGuard/commits/master/worldguard-legacy/src/main/java/com/sk89q/worldguard/bukkit/WorldGuardPlugin.java), it seems like ~~the WG team~~ @me4502 is slowly pulling out pieces of the legacy WG to use elsewhere (maybe not the best practice, but who am I to judge).

FAWE seems to compile against legacy WG regardless of any breaking changes being made. The issue you're experiencing actually comes from July 19 where the getRegionManager() method was [re]moved from WorldGuardPlugin.java in sk89q/WorldGuard@6c5319f355078c26569ac92dd351f26eb29a9458

Until the new WG comes out of beta, the proper solution for you is to downgrade FAWE. The same goes for anyone using the latest beta builds of WG. Frankly, no-one should be using them until it's out of beta. Once it is, other developers and myself will want to update using this guide: https://worldguard.enginehub.org/en/latest/developer/regions/managers/?highlight=regionmanager

**Tl;dr:** Downgrade FAWE or use the last compatible build of the official WG which _should_ be: http://builds.enginehub.org/job/worldguard/10260

**Comment by me4502** [comment]
The worldguard betas are recommended if 1.13 is being used. It’s still a beta as worldedit is a beta, which is still a beta as sponge and forge for 1.13 aren’t finished yet.

**Comment by FlyingPikachu** [comment]
Whoops, updated my first sentence to be a little more clear. Thanks!

It’s unfortunate that legacy (pre-1.13) WG is being chopped up to the point that it’s breaking projects which depend on it. Solution for OP is still to downgrade to earlier builds of WG or FAWE.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
